File: ScheduleViewer.Domain/Entities/AnimeEntity.cs
namespace ScheduleViewer.Domain.Entities;

/// <summary>
/// Entity - アニメ情報
/// </summary>
/// <param name="registeredAnnict">Annictに登録されているか</param>
/// <param name="title">タイトル</param>
/// <param name="seasonName">制作シーズン</param>
/// <param name="seasonYear">制作年</param>
/// <param name="officialSiteUrl">公式サイト</param>
/// <param name="wikipediaUrl">Wikipedia URL</param>
/// <param name="episodesCount">エピソード数</param>
/// <param name="cast">キャスト</param>
/// <param name="thumbnail">サムネイル</param>
/// <param name="part">パート数</param>
/// <param name="subTitle">サブタイトル</param>
/// <param name="watchedFrom">視聴先</param>
/// <param name="caption">概要(話数別)</param>
/// <remarks>
/// こちらはViewModelと紐づくEntityなので、コントロールの増減に応じて調整可能
/// </remarks>
public sealed class AnimeEntity(
    bool registeredAnnict,
    string title, 
    string seasonName, 
    string seasonYear,
    string officialSiteUrl, 
    string wikipediaUrl, 
    string episodesCount,
    string cast,
    string thumbnail,
    string part,
    string subTitle,
    string watchedFrom,
    string caption)
{
    /// <summary> Annictに登録されているか </summary>
    public bool RegisteredAnnict { get; } = registeredAnnict;

    /// <summary> タイトル </summary>
    public string? Title { get; } = title;

    /// <summary> 制作シーズン </summary>
    public string? SeasonName { get; } = seasonName;

    /// <summary> 制作年 </summary>
    public string SeasonYear { get; } = seasonYear;

    /// <summary> 公式サイト </summary>
    public string? OfficialSiteUrl { get; } = officialSiteUrl;

    /// <summary> Wikipedia URL </summary>
    public string? WikipediaUrl { get; } = wikipediaUrl;

    /// <summary> エピソード数 </summary>
    public string EpisodesCount { get; } = episodesCount;

    /// <summary> キャスト </summary>
    public string? Cast { get; } = cast;

    /// <summary> サムネイル </summary>
    public string? Thumbnail { get; } = thumbnail;

    /// <summary> パート数 </summary>
    public string Part { get; } = part;

    /// <summary> サブタイトル </summary>
    public string SubTitle { get; } = subTitle;

    /// <summary> 視聴先 </summary>
    public string WatchedFrom { get; } = watchedFrom;

    /// <summary> 概要(話数別) </summary>
    public string Caption { get; } = caption;

    /// <summary>
    /// シーズン(表示用)
    /// </summary>
    /// <returns>シーズン</returns>
    public string DisplaySeason()
    {
        if (RegisteredAnnict)
        {
            return $"{this.SeasonYear}年 {this.SeasonName}";
        }
        else
        {
            return string.Empty;
        }
    }

    /// <summary>
    /// パート(表示用)
    /// </summary>
    /// <returns>パート</returns>
    public string DisplayPart()
    {
        if (RegisteredAnnict)
        {
            return $"{this.Part} / {this.EpisodesCount}";

        }
        else
        {
            return this.Part;
        }
    }
}

File: schedule-viewer-api/domain/src/main/java/com/scheduleviewer/domain/entity/AnimeEntity.java
package com.scheduleviewer.domain.entity;

/**
 * Entity - アニメ情報
 */
public final class AnimeEntity {

    private final boolean registeredAnnict;
    private final String title;
    private final String seasonName;
    private final String seasonYear;
    private final String officialSiteUrl;
    private final String wikipediaUrl;
    private final String episodesCount;
    private final String cast;
    private final String thumbnail;
    private final String part;
    private final String subTitle;
    private final String watchedFrom;
    private final String caption;

    public AnimeEntity(
            boolean registeredAnnict,
            String title,
            String seasonName,
            String seasonYear,
            String officialSiteUrl,
            String wikipediaUrl,
            String episodesCount,
            String cast,
            String thumbnail,
            String part,
            String subTitle,
            String watchedFrom,
            String caption) {
        this.registeredAnnict = registeredAnnict;
        this.title            = title;
        this.seasonName       = seasonName;
        this.seasonYear       = seasonYear;
        this.officialSiteUrl  = officialSiteUrl;
        this.wikipediaUrl     = wikipediaUrl;
        this.episodesCount    = episodesCount;
        this.cast             = cast;
        this.thumbnail        = thumbnail;
        this.part             = part;
        this.subTitle         = subTitle;
        this.watchedFrom      = watchedFrom;
        this.caption          = caption;
    }

    /** Annictに登録されているか */
    public boolean isRegisteredAnnict() { return registeredAnnict; }

    /** タイトル */
    public String getTitle() { return title; }

    /** 制作シーズン */
    public String getSeasonName() { return seasonName; }

    /** 制作年 */
    public String getSeasonYear() { return seasonYear; }

    /** 公式サイト */
    public String getOfficialSiteUrl() { return officialSiteUrl; }

    /** Wikipedia URL */
    public String getWikipediaUrl() { return wikipediaUrl; }

    /** エピソード数 */
    public String getEpisodesCount() { return episodesCount; }

    /** キャスト */
    public String getCast() { return cast; }

    /** サムネイル */
    public String getThumbnail() { return thumbnail; }

    /** パート数 */
    public String getPart() { return part; }

    /** サブタイトル */
    public String getSubTitle() { return subTitle; }

    /** 視聴先 */
    public String getWatchedFrom() { return watchedFrom; }

    /** 概要 */
    public String getCaption() { return caption; }

    /** シーズン(表示用) */
    public String displaySeason() {
        return registeredAnnict ? seasonYear + "年 " + seasonName : "";
    }

    /** パート(表示用) */
    public String displayPart() {
        return registeredAnnict ? part + " / " + episodesCount : part;
    }
}
